.github/workflows/test.yml

```
@@ -0,0 +1,190 @@
+name: Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - develop
+      - main
+
+env:
+  DANGER_GITHUB_API_TOKEN: ${{ github.token }}
+  DOCKER_REGISTRY_HOST: ${{ secrets.DOCKER_REGISTRY_HOST }}
+  DOCKER_REGISTRY_USERNAME: ${{ github.repository_owner }}
+  DOCKER_REGISTRY_TOKEN: ${{ secrets.DOCKER_REGISTRY_TOKEN }}
+  DOCKER_IMAGE: ${{ github.repository }}
+  NODE_VERSION: 16
+
+  # Set the default docker-compose file
+  COMPOSE_FILE: docker-compose.test.yml
+
+  # Set the default CI value despite the Github Action default value
+  CI: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build Docker image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env BRANCH_TAG
+        uses: nimblehq/branch-tag-action@1
+
+      - name: Login to Docker registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.DOCKER_REGISTRY_HOST }}
+          username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+          password: ${{ env.DOCKER_REGISTRY_TOKEN }}
+
+      - name: Pull Docker image
+        if: ${{ env.BRANCH_TAG != 'latest' && env.BRANCH_TAG != 'develop' }}
+        run: docker-compose pull test || true
+
+      - name: Build Docker image
+        run: bin/docker-prepare && docker-compose build
+
+      - name: Push Docker image
+        run: docker-compose push test
+
+  linting:
+    name: Linting
+    runs-on: ubuntu-latest
+    needs: build
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env BRANCH_TAG
+        uses: nimblehq/branch-tag-action@v1.2
+
+      - name: Login to Docker registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.DOCKER_REGISTRY_HOST }}
+          username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+          password: ${{ env.DOCKER_REGISTRY_TOKEN }}
+
+      - name: Pull Docker image
+        run: docker-compose pull test || true
+
+      - name: Run codebase test
+        run: docker-compose run test bundle exec rspec spec/codebase/codebase_spec.rb --format progress
+
+  unit_tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    needs: build
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env BRANCH_TAG
+        uses: nimblehq/branch-tag-action@1
+
+      - name: Login to Docker registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.DOCKER_REGISTRY_HOST }}
+          username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+          password: ${{ env.DOCKER_REGISTRY_TOKEN }}
+
+      - name: Pull Docker image
+        run: docker-compose pull test || true
+
+      - name: Run unit tests
```

That can besimplified to run all tests:

```suggestion
      - name: Run tests
```